modin/core/execution/unidist/implementations/pandas_on_unidist/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,566 @@
+            PandasDataframeAxisPartition.deploy_func_between_two_axis_partitions,
+            axis,
+            func,
+            f_args,
+            f_kwargs,
+            num_splits,
+            len_of_left,
+            other_shape,
+            *partitions,
+        )
+
+    def _wrap_partitions(self, partitions):
+        """
+        Wrap partitions passed as a list of ``unidist.ObjectRef`` with ``PandasOnUnidistDataframePartition`` class.
+
+        Parameters
+        ----------
+        partitions : list
+            List of ``unidist.ObjectRef``.
+
+        Returns
+        -------
+        list
+            List of ``PandasOnUnidistDataframePartition`` objects.
+        """
+        return [
+            self.partition_type(object_id, length, width, ip)
+            for (object_id, length, width, ip) in zip(*[iter(partitions)] * 4)
+        ]
+
+    def apply(
+        self,
+        func,
+        *args,
+        num_splits=None,
+        other_axis_partition=None,
+        maintain_partitioning=True,
+        lengths=None,
+        manual_partition=False,
+        **kwargs,
+    ):
+        """
+        Apply a function to this axis partition along full axis.
+
+        Parameters
+        ----------
+        func : callable
+            The function to apply.
+        *args : iterable
+            Additional positional arguments to be passed in `func`.
+        num_splits : int, default: None
+            The number of times to split the result object.
+        other_axis_partition : PandasDataframeAxisPartition, default: None
+            Another `PandasDataframeAxisPartition` object to be applied
+            to func. This is for operations that are between two data sets.
+        maintain_partitioning : bool, default: True
+            Whether to keep the partitioning in the same
+            orientation as it was previously or not. This is important because we may be
+            operating on an individual AxisPartition and not touching the rest.
+            In this case, we have to return the partitioning to its previous
+            orientation (the lengths will remain the same). This is ignored between
+            two axis partitions.
+        lengths : list, optional
+            The list of lengths to shuffle the object.
+        manual_partition : bool, default: False
+            If True, partition the result with `lengths`.
+        **kwargs : dict
+            Additional keywords arguments to be passed in `func`.
+
+        Returns
+        -------
+        list
+            A list of `PandasOnUnidistDataframeVirtualPartition` objects.
+
+        Notes
+        -----
+        In older versions of Modin, ``args`` was passed internally as ``kwargs["args"]``, and
+        deserialization was handled in a special case in ``deploy_unidist_func``, making control flow
+        difficult to follow. All deserialization is still handled in ``deploy_unidist_func``, but
+        in a more direct fashion.
```

Comment:
feels like we don't need this note anymore

Reply:
removed